mlflow/genai/scorers/base.py

```
@@ -1,21 +1,194 @@
+        )
+
+        # Check if this is a decorator scorer
+        if hasattr(self, "_original_func") and self._original_func:
+            # Decorator scorer - extract and store source code
+            source_info = self._extract_source_code_info()
+            serialized.call_source = source_info.get("call_source")
+            serialized.call_signature = source_info.get("call_signature")
+            serialized.original_func_name = source_info.get("original_func_name")
+        else:
+            # BuiltInScorer overrides `model_dump`, so this is neither a builtin scorer nor a decorator scorer
+            raise MlflowException.invalid_parameter_value(
+                f"Unsupported scorer type: {self.__class__.__name__}. "
+                f"Scorer serialization only supports:\n"
+                f"1. Builtin scorers (from mlflow.genai.scorers.builtin_scorers)\n"
+                f"2. Decorator-created scorers (using @scorer decorator)\n"
+                f"Direct subclassing of Scorer is not supported for serialization. "
+                f"Please use the @scorer decorator instead."
+            )
+
+        return asdict(serialized)
+
+    def _extract_source_code_info(self) -> dict:
+        """Extract source code information for the original decorated function."""
+        from mlflow.genai.scorers.utils import extract_function_body
+
+        result = {"call_source": None, "call_signature": None, "original_func_name": None}
+
+        # Extract original function source
+        call_body, _ = extract_function_body(self._original_func)
+        result["call_source"] = call_body
+        result["original_func_name"] = self._original_func.__name__
+
+        # Store the signature of the original function
+        try:
+            result["call_signature"] = str(inspect.signature(self._original_func))
+        except Exception:
+            _logger.warning("Failed to extract original function signature for scorer")
```

Will this cause error inside monitoring job? If so, let's fail fast.